### ISSUE TITLE
Line numbers disappear on empty lines if fill column indicator enabled and the indicator is outside the visible buffer area

### ISSUE THREAD

**Issue by katusk** [alpaker/fill-column-indicator#71]
#### Description

Line numbers disappear on empty lines if fill column indicator enabled and the indicator is outside the visible buffer area.
#### Reproduction Guide
- Start a graphical emacs by executing `emacs -Q`
- Open a new buffer, type some text leaving some lines blank
- Enable linum-mode by `M-x linum-mode`
- Fetch, then load fill-column-indicator.el by `M-x load-file`
- Enable fci-mode by `M-x fci-mode`
- Decrease the window width, so that the fill column indicator is not shown
#### Observed Behaviour

On all the empty lines, the line numbers disappear -- except on that empty line where your cursor might rest.
#### Expected Behaviour

Line numbers should be shown on all lines, including empty lines independent of the visibility of the fill column indicator.
#### Linked Issues
- [Spacemacs Issue #6708](https://github.com/syl20bnr/spacemacs/issues/6708)
- [GNU bug report logs - #24638](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=24638)
#### System Information

```
In GNU Emacs 24.5.1 (x86_64-w64-mingw32)
 of 2015-05-16 on KAEL
Windowing system distributor `Microsoft Corp.', version 6.3.9600
Configured using:
 `configure --prefix=/z/emacs --host=x86_64-w64-mingw32
 --target=x86_64-w64-mingw32 --build=x86_64-w64-mingw32 --with-wide-int
 --with-jpeg --with-xpm --with-png --with-tiff --with-rsvg --with-xml2
 --with-gnutls --with-sound=yes --with-file-notification=yes
 --without-dbus --without-imagemagick 'CFLAGS=-O3 -fomit-frame-pointer
 -g0 -pipe' 'LDFLAGS=-static-libgcc -static-libstdc++ -static -s
 -Wl,-s''

Important settings:
  value of $LANG: HUN
  locale-coding-system: cp1250

Major mode: Lisp Interaction

Minor modes in effect:
  fci-mode: t
  linum-mode: t
  tooltip-mode: t
  electric-indent-mode: t
  mouse-wheel-mode: t
  tool-bar-mode: t
  menu-bar-mode: t
  file-name-shadow-mode: t
  global-font-lock-mode: t
  font-lock-mode: t
  blink-cursor-mode: t
  auto-composition-mode: t
  auto-encryption-mode: t
  auto-compression-mode: t
  line-number-mode: t
  transient-mark-mode: t

Recent messages:
For information about GNU Emacs and the GNU system, type C-h C-a.
Linum mode enabled
Loading f:/home/katusk/.emacs.d/elpa/fill-column-indicator-20151030.1233/fill-column-indicator.el (source)...done
Fci mode enabled

Load-path shadows:
None found.

Features:
(shadow sort gnus-util mail-extr emacsbug message format-spec rfc822 mml
easymenu mml-sec mm-decode mm-bodies mm-encode mail-parse rfc2231
mailabbrev gmm-utils mailheader sendmail rfc2047 rfc2045 ietf-drums
mm-util help-fns mail-prsvr mail-utils fill-column-indicator easy-mmode
linum time-date tooltip electric uniquify ediff-hook vc-hooks
lisp-float-type mwheel dos-w32 ls-lisp w32-common-fns disp-table w32-win
w32-vars tool-bar dnd fontset image regexp-opt fringe tabulated-list
newcomment lisp-mode prog-mode register page menu-bar rfn-eshadow timer
select scroll-bar mouse jit-lock font-lock syntax facemenu font-core
frame cham georgian utf-8-lang misc-lang vietnamese tibetan thai
tai-viet lao korean japanese hebrew greek romanian slovak czech european
ethiopic indian cyrillic chinese case-table epa-hook jka-cmpr-hook help
simple abbrev minibuffer nadvice loaddefs button faces cus-face macroexp
files text-properties overlay sha1 md5 base64 format env code-pages mule
custom widget hashtable-print-readable backquote make-network-process
w32notify w32 multi-tty emacs)

Memory information:
((conses 16 77152 5390)
 (symbols 56 17633 0)
 (miscs 48 129 83)
 (strings 32 11181 3730)
 (string-bytes 1 288751)
 (vectors 16 9069)
 (vector-slots 8 378181 17509)
 (floats 8 58 311)
 (intervals 56 273 26)
 (buffers 960 11))
```

**Comment by xuhdev** [comment]
I have removed this package 'cause it causes more issues than benefits. I'm now using `highlight-beyond-fill-column.el` from http://alioth.debian.org/snapshots.php?group_id=30060 as a replacement.

**Comment by alpaker** [comment]
Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
